connectors/sources/dropbox.py

```diff
@@ -684,15 +817,33 @@ async def get_docs(self, filtering=None):
         Yields:
             dictionary: dictionary containing meta-data of the files.
         """
-        async for document, attachment in self._fetch_files_folders(
-            path=self.dropbox_client.path
-        ):
-            if document["type"] == FILE:
-                yield document, partial(self.get_content, attachment=attachment)
-            else:
-                yield document, None
-
-        async for document, attachment in self._fetch_shared_files():
-            yield document, partial(
-                self.get_content, attachment=attachment, is_shared=True
-            )
+        if filtering and filtering.has_advanced_rules():
+            advanced_rules = filtering.get_advanced_rules()
+            for rule in advanced_rules:
+                self._logger.debug(f"Fetching files using advanced sync rule: {rule}")
+
+                async for document, attachment in self.advanced_sync(rule=rule):
+                    if document["type"] == FILE:
+                        if document.get("url"):
```

Comment:
would `if document.get("url")` also be true for documents that come out of `self._fetch_shared_files()`? If so, I think this new block could be DRY'd up with the below section. Like pulling out a separate function like:
```
def document_tuple(self, document, attachment)
  if document["type"] == FILE:
    if document.get("url"):
      yield document, partial(self.get_content, attachment=attachment, is_shared=True)
    else:
       yield document, partial( self.get_content, attachment=attachment)
   else:
     yield document, None
     
     
...
      async for document, attachment in self.advanced_sync(rule=rule):
        yield document_tuple(document, attachment)
  else:
    async for document, attachment in self._fetch_files_folders(path=self.dropbox_client.path):
      yield document_tuple(document, attachment)
```

Reply:
Yes, Thanks for the suggestion! I think making this code snippet DRY'd would be great 👍